src/pages/posts/[postid].astro

```diff
@@ -41,10 +42,22 @@ const html = sanitizeHtml(await marked.parse(content.body), {
 const showCopyLinkWarning = Astro.url.searchParams.get('created') === content.edit_key;
 const url = Astro.url.href.split('?')[0];
 const editURL = url + `/edit?key=${content.edit_key}`;
----
 
+const canViewAdultContent = Astro.url.searchParams.get('adult');
+const showAdultWarning = !canViewAdultContent && content.adult_content;
+---
 <BaseLayout title={`${content.title || 'Untitled'}`}>
     <Seo title={content.title} description={trimmedText} />
+
+    <Modal client:only="preact" isOpen={true} hidePage={true}>
+        <h1>Heads up</h1>
+        <p>The following post contains adult content. By proceeding, you agree that you are at least 18 years old.</p>
+        <div class="modalButtons">
+            <form method="dialog"><button>Ok</button></form>
```

Comment:
it's a little strange to have an "ok" button and an "x" button on this dialog, maybe remove the close button?